packages/crud/src/vaadin-crud.js

```diff
@@ -824,7 +833,8 @@ class Crud extends SlotMixin(ControllerMixin(ElementMixin(ThemableMixin(PolymerE
   /** @private */
   __onDomChange(addedNodes) {
     // TODO: restore default button when a corresponding slotted button is removed.
-    // Consider creating a controller to reuse custom helper logic from FieldMixin.
+    // This would require us to detect where to insert a button after teleporting it,
+    // which happens after opening a dialog and closing it (default editor position).
```

Comment:
Updated the comment to clarify it. I've tried to implement this, but it ended up with adding more complexity to the "teleportation" logic. Also, adding a button and then removing it is rather an edge case.